src/Parallel/Main.hpp

```diff
@@ -637,9 +641,26 @@ void Main<Metavariables>::
 
 template <typename Metavariables>
 void Main<Metavariables>::execute_next_phase() {
-  current_phase_ = Metavariables::determine_next_phase(
+  ASSERT(Parallel::Phase::Exit != current_phase_,
```

Comment:
should we promote this and the `ASSERT` below to `ERROR`s to avoid weird bugs in production? I'm just thinking it's a cheap thing to check and might make life a bit easier on occasion